### ISSUE TITLE
Замечания

### ISSUE THREAD

**Issue by maximpetrov** [Danil891/second_task#1]
1. При разборе аргументов, Вы не проверяете корректность их задания: Вы проигнорируете лишние аргументы и не проверяете наличие обязательных аргументов (`word` и `filename`). Кроме того, непонятно, почему Вы ищете `word` в любом месте, а `filename` только в конце списка.
1. https://github.com/Danil891/second_task/blob/32a4964fa7221e56564286480ca149499ee8a2e2/src/Main.java#L18 Зачем этот вызов?
1. Результат нужно выводить построчно.
1. У Вас нет реализации для случаев `-i -r` и `-i -v -r`.
1. Не понятно, зачем Вам метод `sOut`, который лишний раз читает файл вместо того, чтобы сразу запоминать строки.
1. Видимо, Вы не совсем верно поняли задание, но оно не предполагает, что `word` - это выделенное слово, искать предполагается просто подстроку.
1. У Вас очень много практически идентичного кода. Все эти 6 (а с учётом пункта 4 даже 8) функций можно легко объединить в одну.

**Comment by Danil891** [comment]
Да, я понял, что задание просит отдельно искать полностью слово, а -r уже любое совпадение по заданному шаблону.
Случаи -r  -i  и  -r -v -i я посчитал необязательными, т. к. уже задано регулярное выражение и поиск должен отталкиваться от него.
При объединении функций лучше воспользоваться Pattern & Matcher и получается все время реализовывать поиск по 
 различным регулярным выражениям, которые поступают на вход?

**Comment by maximpetrov** [comment]
Даже при заданном регулярном выражении искать его можно без учёта регистра, для этого есть специальный флаг.
Для объединения можно воспользоваться тем, что из обычного слова, которое нужно найти, легко делается регулярное выражение, которое это слово ищет.

**Comment by Danil891** [comment]
Я сейчас уже переделал все через Pattern & Matcher
И сейчас переделываю тесты и подключаю args4, чтобы переделать все с помощь аннотаций
